app/services/discovery_engine/sync/locking.rb

```diff
@@ -63,14 +63,14 @@ def newer_than_remote?(content_id, payload_version:)
 
     # Gets the latest synced version for a document from Redis
     def latest_synced_version(content_id)
-      RedisPool.with do |redis|
+      Rails.application.config.redis_pool.with do |redis|
```

Comment:
This is much better, and it's much more obvious where the value comes from.

Perhaps this commit can be fixed up, so that all references to `RedisPool` are removed from the commit history, so a future developer looking to see how the connection pool was configured doesn't accidentally trip over this.